include/dlaf/common/index2d.h

```diff
@@ -156,24 +152,15 @@ class Size2D : public internal::basic_coords<IndexT> {
 /// @tparam Tag for strong-typing
 template <typename IndexT, class Tag>
 class Index2D : public internal::basic_coords<IndexT> {
-  using base = internal::basic_coords<IndexT>;
+  using BaseT = internal::basic_coords<IndexT>;
 
 public:
+  using BaseT::basic_coords;
+
   using IndexType = IndexT;
 
   /// Create an invalid 2D coordinate
-  Index2D() noexcept : internal::basic_coords<IndexT>(-1, -1) {}
-
-  Index2D(base const& coords) noexcept : base(coords) {}
-
-  /// Create a valid 2D coordinate
-  /// @param row index of the row
-  /// @param col index of the column
-  /// @throw std::invalid_argument if row < 0 or col < 0
-  Index2D(IndexT row, IndexT col) : internal::basic_coords<IndexT>(row, col) {
-    if (!internal::basic_coords<IndexT>::isValid())
-      throw std::invalid_argument("indices are not valid (negative).");
-  }
```

Comment:
@teonnik 

I've just realized that it was possible to create an Index2D with "invalid" coords via `Index2D(base const& coords) noexcept` but it was checked to be valid via `Index2D(IndexT row, IndexT col)`.

I thought it was worth mention it, since the behavior may have changed (maybe it was neither tested).

Reply:
I removed `Index2D(base const& coords) noexcept` constructor later in the PR.